### ISSUE TITLE
[feature] render full handles for mentions

### ISSUE THREAD

**Issue by mirabilos** [superseriousbusiness/gotosocial#2272]
### Is your feature request related to a problem ?

Sometimes, people mention four different `@name` in one post. I also tend to remember the whole string, or a combination of a bit of the name and a bit of the domain.

### Describe the solution you'd like.

I’d like for statuses shown to me by clients to always use the full `@name@domain` form and not the short `@name` form.

I’m not _entirely_ sure that this is not (also) a client issue, but in my Semaphore home TL I’m seeing a mix of full and shortened mentions, so it seems the _sending_ side sets them?

So therefore, I would like…

1. for GtS as sending side to render them as full handles
2. for GtS as receiving side, if at all possible, expand them to the full handles

### Describe alternatives you've considered.

asking the client developers, but that’d duplicate effort and have a lower chance of success

**Comment by tsmethurst** [comment]
>  the sending side sets them

That's right. If you federate with any misskey instances, you'll see that they always (I believe) set the full handle in their outgoing messages. We just display what clients send us, without doing anything at this point beyond light sanitization.

Changing this on GtS's side would require messing about with incoming messages either as they come in, or when we serve them to API clients. This involves parsing and replacing the HTML of specific mentions in a text either once or on-the-fly every time.

So, not a trivial change unfortunately! Definitely see the utility in doing this, but it might be a while before (if!) we get around to it, or we might only get around to it as one part of a larger rewrite / feature. Will bear it in mind though, thanks for opening.

**Comment by mirabilos** [comment]
> We just display what clients send us, without doing anything at this point beyond light sanitization.
>
> Changing this on GtS's side would require

Yeah, that’s fine, that’s why this is just a feature request.

Is the other part (writing the full handle when posting *from* GtS) easier? That would only help with a subset of posts, but perhaps a good start and it could spread from there.

But, again, no hurries, you’ve got a release to go out at the moment.

**Comment by tsmethurst** [comment]
> Is the other part (writing the full handle when posting from GtS) easier?

That part is easy yeah

**Comment by daenney** [comment]
We've migrated to Codeberg. Your issue has been imported as https://codeberg.org/superseriousbusiness/gotosocial/issues/2272.
